File: src/com/java/transactions/Transactions.java
package com.java.transactions;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Scanner;

public class Transactions {
    public static void main(String[] args) {
        System.out.println("Executing Java tests...");

        try {
            String httpResponse = httpRequest("https://jsonplaceholder.typicode.com/posts", "GET", "");
            System.out.println("HTTP Response: " + httpResponse);

            executeCreditLeg(100);
            executeDebitLeg(50);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static String httpRequest(String urlString, String method, String payload) throws IOException {
        URL url = new URL(urlString);
        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
        connection.setRequestMethod(method);
        connection.setRequestProperty("Content-Type", "application/json");

        if ("POST".equals(method) || "PUT".equals(method)) {
            connection.setDoOutput(true);
            connection.getOutputStream().write(payload.getBytes("UTF-8"));
        }

        Scanner scanner = new Scanner(connection.getInputStream());
        StringBuilder response = new StringBuilder();
        while (scanner.hasNextLine()) {
            response.append(scanner.nextLine());
        }
        scanner.close();

        return response.toString();
    }

    public static void executeCreditLeg(int amount) {
        System.out.println("Credit leg executed with amount: " + amount);
    }

    public static void executeDebitLeg(int amount) {
        System.out.println("Debit leg executed with amount: " + amount);
    }
}

File: src/typescript/transactions.ts
// Importing necessary modules
import axios from 'axios';

class Transactions {
    public static async main(): Promise<void> {
        console.log("Executing TypeScript tests...");

        const args = process.argv.slice(2);
        const amountCredit = parseInt(args[0], 10) || 100; // Default to 100 if not provided
        const amountDebit = parseInt(args[1], 10) || 50; // Default to 50 if not provided

        try {
            const httpResponse = await Transactions.httpRequest("https://jsonplaceholder.typicode.com/posts", "GET", "");
            console.log("HTTP Response: " + httpResponse);

            Transactions.executeCreditLeg(amountCredit);
            Transactions.executeDebitLeg(amountDebit);
        } catch (error) {
            console.error('Error:', error);
        }
    }

    private static async httpRequest(url: string, method: string, payload: string): Promise<string> {
        // Configure request based on method
        const options = {
            method: method,
            url: url,
            headers: { 'Content-Type': 'application/json' },
            data: (method === "POST" || method === "PUT") ? payload : null
        };

        // Axios automatically handles request method and data
        try {
            const response = await axios(options);
            return JSON.stringify(response.data);  // Converting response data to string for consistent output
        } catch (error) {
            console.error('HTTP request failed:', error);
            throw error;  // Re-throwing the error for upstream handling
        }
    }

    private static executeCreditLeg(amount: number): void {
        console.log("Credit leg executed with amount: " + amount);
    }

    private static executeDebitLeg(amount: number): void {
        console.log("Debit leg executed with amount: " + amount);
    }
}

// Running the main function of Transactions class only if executed directly
if (require.main === module) {
    Transactions.main();
}
